### PR TITLE
61 alpine mariadb

### PR DESCRIPTION
**Pull request by rtprio**
Put this in a PR to see how it fares in CI.

### CODE REVIEW

**Review by Schnitzel — CHANGES_REQUESTED** [review]

File: images/alpine-mariadb10/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
+FROM alpine
+
+MAINTAINER amazee.io
+ENV BACKUPS_DIR="/mnt/backups"
+
+RUN touch /amazeeio
+
+RUN apk add --update --no-cache bash ca-certificates curl gzip tar unzip wget && rm -rf /var/cache/apk/*
```
[the remark this answers is not in the document]
Comment (rtprio) [reply images/alpine-mariadb10/Dockerfile]: removed.

File: images/alpine-mariadb10/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+COPY docker-entrypoint.sh /
+COPY my.cnf /etc/mysql/my.cnf
+
+
+
+WORKDIR /var/lib/mysql
+VOLUME /var/lib/mysql
+EXPOSE 3306
+
+ENTRYPOINT ["/docker-entrypoint.sh"]
```
[the remark this answers is not in the document]
Comment (rtprio) [reply images/alpine-mariadb10/Dockerfile]: ✅

**Comment by rtprio** [comment]
@Schnitzel not sure why the test is still failing.

File: images/alpine-mariadb10/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,27 @@
+FROM alpine
+
```
[the remark this answers is not in the document]
Comment (rtprio) [reply images/alpine-mariadb10/Dockerfile]: this has been revised.

File: images/alpine-mariadb10/test.sh [hunk 4]
```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (rtprio) [reply images/alpine-mariadb10/test.sh]: removed

File: images/alpine-mariadb10/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+ENV TMPDIR=/tmp TMP=/tmp HOME=/home ENV=/home/.bashrc BASH_ENV=/home/.bashrc
+
+MAINTAINER amazee.io
+ENV BACKUPS_DIR="/var/lib/mysql/backup"
+
+RUN apk add --no-cache bash mariadb-client mariadb mariadb-common pwgen && rm -rf /var/cache/apk/*
+
+RUN set -xe; \
+    mkdir -p      /var/run/mysqld /var/lib/mysql /etc/mysql /etc/mysql/conf.d /etc/mysql/ "${BACKUPS_DIR}"; \
+    chgrp -R root /var/run/mysqld /var/lib/mysql /etc/mysql /etc/mysql/conf.d /etc/mysql/ "${BACKUPS_DIR}"; \
```
[the remark this answers is not in the document]
Comment (rtprio) [reply images/alpine-mariadb10/Dockerfile]: fixed in 14b2cfb091a93e2c97c3dbeb4080930c93e656d4

File: images/alpine-mariadb10/docker-entrypoint.sh [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+	echo "[i] MySQL data directory not found, creating initial DBs"
+
+	mysql_install_db --skip-name-resolve
+
+	if [ "$MYSQL_ROOT_PASSWORD" = "" ]; then
+		MYSQL_ROOT_PASSWORD=`pwgen 16 1`
+		echo "[i] MySQL root Password: $MYSQL_ROOT_PASSWORD"
+	fi
+
+	MYSQL_DATABASE=${MYSQL_DATABASE:-""}
```
[the remark this answers is not in the document]
Comment (rtprio) [reply images/alpine-mariadb10/docker-entrypoint.sh]: fixed in eb5f743267e805f2a294ce4dd32128731e6887e8

this still feels a little awkward as mariadb haven't changed all the mysql references.

**Review by Schnitzel — CHANGES_REQUESTED** [review]
Nice progress :)

Now two things
1. I would like that the image is just called `mariadb` instead of `alpine-mariadb10`. But this will conflict with the service called `mariadb` in Lagoon, as they have the same Docker image names (had that issue with elasticsearch as a based image and as a service). So maybe let's rename the service `mariadb` into `api-db` and then the base image can be called `mariadb`
2. Please also add the mariadb backup script: https://github.com/amazeeio/lagoon/blob/master/images/centos7-mariadb10/mysql-backup.sh

**Review by Schnitzel — CHANGES_REQUESTED** [review]
nice!

now the last thing, let's actually remove all of `centos7`, `centos7-mariadb10`, `centos7-mariadb10-drupal`  and so we can check that we really don't need them anymore

**Comment by rtprio** [comment]
The alpine package is mariadb 10.1.  The centos packages is mariadb 10.2.
The views in the API use a subquery in the from clause, which isn't supported until 10.2.

`ERROR: 1349  View's SELECT contains a subquery in the FROM clause`

**Comment by Schnitzel** [comment]
@rtprio found a fix: https://github.com/amazeeio/lagoon/pull/196/commits/0c32eb3bac791d325c438aa6cc9226590cbdfa05
based on https://stackoverflow.com/a/8428703/2397973

File: images/mariadb/docker-entrypoint.sh [hunk 7]
```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env bash
+
+set -e
+set -x
```
Comment (Schnitzel) [inline images/mariadb/docker-entrypoint.sh:4]: can you add here the 
```
set -eo pipefail
shopt -s nullglob
```
from the original docker-entrypoint?

**Review by Schnitzel — DISMISSED** [review]
one small thing!

**Comment by Schnitzel** [comment]
ugh, it fails now with 
```
/lagoon/entrypoints.sh: /lagoon/entrypoints/90-mariadb-entrypoint: line 1: shopt: not found
```

File: images/mariadb/docker-entrypoint.sh (hunk 7 above)
Comment (rtprio) [reply images/mariadb/docker-entrypoint.sh:4]: shopt is unsupported.

**Review by Schnitzel — APPROVED** [review]
niiiiice!

**Pull request merged**